### PR TITLE
improve affine bounding box reference helper

### PR DESCRIPTION
**Pull request by pmeier**
Improvements for `reference_affine_bounding_boxes_helper`:

1. Pass a `datapoints.BoundingBoxes` directly rather than a `torch.Tensor` with explicit metadata
2. Automatic wrapping of the output into `datapoints.BoundingBoxes`
3. Support for `expand=True`, which is needed for the bounding box correctness tests of `F.rotate` and `F.resize`

cc @vfdev-5

### CODE REVIEW

File: test/test_transforms_v2_refactored.py [hunk 1]
```diff
@@ -339,45 +340,95 @@ def assert_warns_antialias_default_value():
...
-                np.min(transformed_points[:, 0]).item(),
-                np.min(transformed_points[:, 1]).item(),
-                np.max(transformed_points[:, 0]).item(),
-                np.max(transformed_points[:, 1]).item(),
-            ],
-            dtype=bbox_xyxy.dtype,
-        )
-        out_bbox = F.convert_bounding_box_format(
-            out_bbox, old_format=datapoints.BoundingBoxFormat.XYXY, new_format=format, inplace=True
+def bounding_boxes_reference_handle_batching(reference_fn):
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:343]: Why does this need to be a decorator or even a separate function from `reference_affine_bounding_boxes_helper` ?

File: torchvision/transforms/v2/functional/_geometry.py [hunk 2]
```diff
@@ -803,8 +803,8 @@ def _affine_bounding_boxes_with_expand(
         tr = torch.amin(new_points, dim=0, keepdim=True)
         # Translate bounding boxes
         out_bboxes.sub_(tr.repeat((1, 2)))
-        # Estimate meta-data for image with inverted=True and with center=[0,0]
-        affine_vector = _get_inverse_affine_matrix([0.0, 0.0], angle, translate, scale, shear)
+        # Estimate meta-data for image with inverted=True
+        affine_vector = _get_inverse_affine_matrix(center, angle, translate, scale, shear)
```
Comment (pmeier) [inline torchvision/transforms/v2/functional/_geometry.py:807]: See https://github.com/pytorch/vision/pull/7723#discussion_r1295888903.

File: test/test_transforms_v2_refactored.py [hunk 3]
```diff
@@ -351,45 +351,70 @@ def assert_warns_antialias_default_value():
         yield
 
 
-def reference_affine_bounding_boxes_helper(bounding_boxes, *, format, canvas_size, affine_matrix):
-    def transform(bbox):
+def reference_affine_bounding_boxes_helper(bounding_boxes, *, affine_matrix, expand=False):
+    def affine_bounding_boxes(bounding_boxes):
```
Comment (vfdev-5) [inline test/test_transforms_v2_refactored.py:355]: Nit: here method name is unclear what it does as `affine_bounding_boxes` reads as a noun and not a verb

**Review by vfdev-5 — APPROVED** [review]
LGTM, although it is a bit complicated to trace all edge cases this updates can bring.
Let's on the CI

File: test/test_transforms_v2_refactored.py (hunk 3 above)
Comment (pmeier) [reply test/test_transforms_v2_refactored.py:355]: This is the same name as our kernel:

https://github.com/pytorch/vision/blob/fae532174e1cde24542d3b357055d8f37b44e326/torchvision/transforms/v2/functional/_geometry.py#L820

**Comment by pmeier** [comment]
After trying to use the new helper for `F.crop` I realized that having a `canvas_size` overwrite is indeed needed (same for `F.pad`). Thus, having `expand` was only sufficient for the correctness tests of `F.rotate` and `F.resize`. I've removed `expand` from the helper again, but kept the functionality for the correctness test for `F.expand`. 

Thus, point 3. in https://github.com/pytorch/vision/pull/7884#issue-1866835837 now changes to

- Add support for `F.rotate` bounding boxes correctness tests with `expand=True`

Point 1. and 2. are still valid.
